examples/name-service/programs/name-service-without-macros/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+
+        if light_accounts.record.owner != ctx.accounts.signer.key() {
+            return err!(CustomError::Unauthorized);
+        }
+
+        light_accounts.record.rdata = new_rdata;
+
+        verify_compressed_accounts(
+            &ctx,
+            inputs.proof,
+            &[light_accounts.record],
+            None,
+            false,
+            None,
+            &crate::ID,
+        )?;
+
+        Ok(())
+    }
+
+    pub fn delete_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, DeleteRecord<'info>>,
+        inputs: Vec<u8>,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let mut account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
+
+        account_infos[0].set_discriminator(NameRecord::discriminator());
+
+        let light_accounts = LightDeleteRecord::try_light_accounts(&account_infos)?;
+
+        if light_accounts.record.owner != ctx.accounts.signer.key() {
+            return err!(CustomError::Unauthorized);
+        }
+
+        verify_compressed_accounts(
+            &ctx,
+            inputs.proof,
+            &[light_accounts.record],
+            None,
+            false,
+            None,
+            &crate::ID,
+        )?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, BorshDeserialize, BorshSerialize)]
+pub enum RData {
+    A(Ipv4Addr),
+    AAAA(Ipv6Addr),
+    CName(String),
+}
+
+impl anchor_lang::IdlBuild for RData {}
+
+impl AsByteVec for RData {
+    fn as_byte_vec(&self) -> Vec<Vec<u8>> {
+        match self {
+            Self::A(ipv4_addr) => vec![ipv4_addr.octets().to_vec()],
+            Self::AAAA(ipv6_addr) => vec![ipv6_addr.octets().to_vec()],
+            Self::CName(cname) => cname.as_byte_vec(),
+        }
+    }
+}
+
+impl Default for RData {
+    fn default() -> Self {
+        Self::A(Ipv4Addr::new(127, 0, 0, 1))
+    }
+}
+
+#[derive(
+    Clone, Debug, Default, AnchorDeserialize, AnchorSerialize, LightDiscriminator, LightHasher,
+)]
+pub struct NameRecord {
+    #[truncate]
+    pub owner: Pubkey,
+    #[truncate]
+    pub name: String,
+    pub rdata: RData,
+}
+
+#[error_code]
+pub enum CustomError {
+    #[msg("No authority to perform this action")]
+    Unauthorized,
+}
+
+#[light_system_accounts]
+#[derive(Accounts, LightTraits)]
+pub struct CreateRecord<'info> {
+    #[account(mut)]
+    #[fee_payer]
+    pub signer: Signer<'info>,
+    #[self_program]
+    pub self_program: Program<'info, crate::program::NameService>,
+    /// CHECK: Checked in light-system-program.
+    #[authority]
+    pub cpi_signer: AccountInfo<'info>,
+}
+
+pub struct LightCreateRecord<'a> {
+    pub record: LightAccount<'a, NameRecord>,
+}
+
+impl<'a> LightAccounts<'a> for LightCreateRecord<'a> {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self> {
+        let record: LightAccount<NameRecord> = LightAccount::from_light_account_info(&accounts[0])?;
```

Comment:
```suggestion
    fn try_light_accounts(account_metas: &'a [LightAccountMeta]) -> Result<Self> {
        l```suggestion
        let record: LightAccount<NameRecord> = LightAccount::from_light_account_info::<NameRecord::DISCRIMINATOR>(&account_metas[0])?;
```
```

Reply:
tbh if we derive `try_light_accounts` in macros wouldn't it be the same if we inject the content of `try_light_accounts` into the instruction or instruction context code directly and omit this level of abstraction.
For non-anchor programs this abstraction is already too highlevel in my opinion.

Reply:
I kicked `try_light_accounts` out from the example without macros.

I will remove the trait and any signs of it in the SDK once I get to the macro changes (another PR).

I hope that's fine for you.